### PR TITLE
[spdlog] Added new fuzzers, modifed existing ones

### PR DESCRIPTION
**Pull request by yooyoo9**
Added new fuzzers to spdlog. Improved overall coverage.

### CODE REVIEW

**Comment by inferno-chromium** [comment]
Please "git pull", "git rebase", "git push --force" so that we only see the changes excluding your last cl. Also, fuzzers should be landed in upstream repo and reviewed by upstream developer, please see https://google.github.io/oss-fuzz/advanced-topics/ideal-integration/

**Comment by inferno-chromium** [comment]
Check travis failures

**Comment by inferno-chromium** [comment]
@yooyoo9 ping!

**Comment by yooyoo9** [comment]
The maintainer would not like to have those fuzzers in the upstream repository. Would it be possible to leave them here?

**Comment by inferno-chromium** [comment]
> The maintainer would not like to have those fuzzers in the upstream repository. Would it be possible to leave them here?

Can you ask and add maintainer here for review. It is fine to have fuzzers here, but we want either the maintainer or your host to do the first round of review on these fuzzers.

**Comment by yooyoo9** [comment]
@gabime @spq can you do a review round on these fuzzers

File: projects/spdlog/fuzz/format_fuzzer.cc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  static std::shared_ptr<spdlog::logger> my_logger;
+  if (!my_logger.get()) {
+    my_logger = spdlog::basic_logger_mt("basic_logger", "/dev/null");
+    spdlog::set_default_logger(my_logger);
+  }
+
+  if (size == 0) {
```
Comment (Google-Autofuzz) [inline projects/spdlog/fuzz/format_fuzzer.cc:48]: This should be at the beginning of the function

File: projects/spdlog/fuzz/format_fuzzer.cc [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  FuzzedDataProvider stream(data, size);
+
+  const unsigned long size_arg = stream.ConsumeIntegral<unsigned long>();
+  const unsigned long  int_arg = stream.ConsumeIntegral<unsigned long>();
+  const char flag = (char)(stream.ConsumeIntegral<unsigned char>());
+  const std::string pattern = stream.ConsumeRandomLengthString();
+  my_formatter_txt = stream.ConsumeRandomLengthString();
+  const std::string string_arg = stream.ConsumeRandomLengthString();
+  const std::string format_string = stream.ConsumeRemainingBytesAsString();
+  // const std::string format_string = stream.ConsumeRandomLengthString(size);
```
Comment (Google-Autofuzz) [inline projects/spdlog/fuzz/format_fuzzer.cc:62]: Please remove this

File: projects/spdlog/fuzz/levels_fuzzer.cc [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#include "spdlog/cfg/env.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
+  static std::shared_ptr<spdlog::logger> my_logger;
+  if (!my_logger.get()) {
+    my_logger = spdlog::basic_logger_mt("basic_logger", "/dev/null");
+    spdlog::set_default_logger(my_logger);
+  }
+
+  if (size == 0) {
```
Comment (Google-Autofuzz) [inline projects/spdlog/fuzz/levels_fuzzer.cc:31]: Please move this at the beginning of the function

File: projects/spdlog/fuzz/levels_fuzzer.cc [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  std::vector<std::string> strings;
+  const unsigned char strsize = stream.ConsumeIntegral<unsigned char>();
+  for(unsigned char i=0; i<strsize; i++){
+    strings.push_back(stream.ConsumeRandomLengthString());
+  }
+  std::vector<char*> argvv; argvv.reserve(strsize);
+  for(unsigned char i=0; i< strsize; ++i)
+    argvv.push_back(const_cast<char*>(strings[i].c_str()));
+  
+  const unsigned char int_arg = strsize;
```
Comment (Google-Autofuzz) [inline projects/spdlog/fuzz/levels_fuzzer.cc:47]: Why declare a new variable?

**Comment by yooyoo9** [comment]
My co-host did the first round of review on the fuzzers.

**Comment by gabime** [comment]
@yooyoo9 I reviewed. Looks good.

**Comment by yooyoo9** [comment]
@inferno-chromium first round of review done.

**Pull request merged**